tests/NuGet.Services.AzureSearch.Tests/SearchService/SearchResponseBuilderFacts.cs

```diff
@@ -357,6 +360,88 @@ public void UsesVersionSpecificDownloadCount()
                 Assert.Equal(4, response.Data[0].DownloadCount);
             }
 
+            [Fact]
+            public void SortDownloadsAscUsingAuxilaryFile()
+            {
+                var mockAuxilaryDataDownloads = new Dictionary<string, long>();
+                var rnd = new Random();
+                for (int i = 0; i < _fiveSearchResults.Count; ++i)
+                {
+                    var newDoc = _fiveSearchResults.Results[i];
+                    newDoc.Document.PackageId += i;
+                    newDoc.Document.TotalDownloadCount += i;
+                    newDoc.Document.Title += i;
+
+                    // Set random download count in the Auxilary file for this package
+                    mockAuxilaryDataDownloads.Add(newDoc.Document.PackageId, rnd.Next(0, int.MaxValue));
+                }
+                
+                var expectedOrder = _fiveSearchResults
+                    .Results
+                    .OrderBy(x => mockAuxilaryDataDownloads[x.Document.PackageId])
+                    .ThenBy(x => x.Document.Created)
+                    .Select(x => x.Document.PackageId);
+
+                _auxiliaryData
+                    .Setup(x => x.GetTotalDownloadCount(It.IsAny<string>()))
+                    .Returns((string packageId) => {
+                        return mockAuxilaryDataDownloads[packageId];
+                    });
```

Comment:
Nit: C# style is to put `{` on a new line. In this case, you can shorten this up a little bit

```suggestion
                    .Returns(packageId => mockAuxilaryDataDownloads[packageId]);
```

Reply:
Same comment for lines 428-431